src/linear_algebra/basis.lean

```diff
@@ -610,6 +627,56 @@ begin
         apply linear_map.map_le_range } } }
 end
 
+/-- Dedekind's linear independence of characters -/
+theorem linear_independent_monoid_hom (G : Type*) [monoid G] (L : Type*) [integral_domain L] :
+  linear_independent L (subtype.val : { f : G → L // is_monoid_hom f } → (G → L)) :=
```

Comment:
Bundled monoid homs now exist in mathlib. That should be used here.

Reply:
done